steward/src/allocation.rs

```diff
@@ -41,124 +30,139 @@ pub async fn allocation_precommit(
 pub async fn get_connections() -> Result<Connections, Error> {
     let config = APP.config();
     let timeout = Duration::from_secs(10);
-    let try_base = AllocationQueryClient::connect(config.cosmos.grpc.clone()).await;
-    let (grpc, contact) = match try_base {
-        Ok(val) => (
-            val,
-            Contact::new(&config.cosmos.grpc, timeout, &config.cosmos.prefix)?,
-        ),
-        Err(e) => {
-            error!(
-                "Failed to access Cosmos gRPC with {:?} and create connections",
-                e
-            );
-            return Err(Error::from(e));
-        }
-    };
-    Ok(Connections { grpc, contact })
+
+    Ok(Connections {
+        allocation_client: AllocationQueryClient::connect(config.cosmos.grpc.clone()).await?,
+        contact: Contact::new(&config.cosmos.grpc, timeout, &config.cosmos.prefix)?,
+        gravity_client: GravityQueryClient::connect(config.cosmos.grpc.clone()).await?,
+    })
 }
 
 pub async fn decide_rebalance(
     tick_range: Vec<somm::TickRange>,
-    cellar_address: H160,
+    cellar_address: String,
 ) -> Result<(), Error> {
-    if std::env::var("CELLAR_DRY_RUN").expect("Expect CELLAR_DRY_RUN var") == "TRUE" {
-        return Ok(());
-    } else {
-        let mut connections = get_connections().await?;
-        let (contact, grpc_client) = (connections.contact, &mut connections.grpc);
+    debug!("deciding rebalance for cellar address {}", cellar_address);
+    debug!("getting eth gas price");
+    let eth_gas_price = match cellars::get_gas_price().await {
+        Ok(gp) => {
+            debug!("eth gas price is {}", gp);
+            gp
+        }
+        Err(err) => {
+            error!("failed to get cellar gas price: {}", err);
+            // TO-DO handle this better
+            0.into()
```

Comment:
Nit: I remember you mentioning you wanted to update this but it can wait for another PR.

Reply:
working on it in another branch!